### ISSUE TITLE
Traversing nested file structure in data

### ISSUE THREAD

**Issue by phughk** [phughk/laerning-tool#10]
The data directory may be nested. Traverse the directory looking for available files.

**Comment by sbshah97** [comment]
Is this open for help? Would love to give it a try. 

I am a beginner at Rust. Might be helpful to see where to look at

**Comment by phughk** [comment]
Hey @sbshah97 ! Absolutely, thats awesome of you!
The data directory is evaluated in here https://github.com/phughk/laerning-tool/blob/main/engine/src/main.rs#L55
So you can have a look at the implementation, and do a scan of the folder for files matching a pattern. Lets say the pattern is "*.xml". Then the resulting list would be relative paths to the provided path and include only the xml files :)
Happy to answer any further questions! And thanks for taking an interest in the project!

**Comment by sbshah97** [comment]
Thanks. And really nice meeting you, I was the one who came to you with the Go Starter Project idea 😄

**Comment by phughk** [comment]
Yes I remember :) It was great meeting you as well, really enjoyed the chat!
